src/Infrastructure.Dapper/Vault/Repositories/CipherRepository.cs

```diff
@@ -14,20 +17,31 @@ namespace Bit.Infrastructure.Dapper.Vault.Repositories;
 
 public class CipherRepository : Repository<Cipher, Guid>, ICipherRepository
 {
-    public CipherRepository(GlobalSettings globalSettings)
+    private readonly IFeatureService _featureService;
+
+    private bool UseFlexibleCollections(ICurrentContext currentContext) =>
+        _featureService.IsEnabled(FeatureFlagKeys.FlexibleCollections, currentContext);
+
+    public CipherRepository(GlobalSettings globalSettings, IFeatureService featureService)
         : this(globalSettings.SqlServer.ConnectionString, globalSettings.SqlServer.ReadOnlyConnectionString)
-    { }
+    {
+        _featureService = featureService;
+    }
 
     public CipherRepository(string connectionString, string readOnlyConnectionString)
         : base(connectionString, readOnlyConnectionString)
     { }
 
-    public async Task<CipherDetails> GetByIdAsync(Guid id, Guid userId)
+    public async Task<CipherDetails> GetByIdAsync(Guid id, Guid userId, ICurrentContext currentContext)
```

Comment:
⚠️ To further my other comment, we should pass results of flag evaluation into the repository from the service layer. The repository merely owns the decision of what database entity to use.